### PR TITLE
Mostly replaced Promise with Async within Products.

### PR DESCRIPTION
**Pull request by elvixpix**

### CODE REVIEW

File: api/db/repositories/productRepository.js [hunk 1]
```diff
@@ -3,18 +3,19 @@ const Prices = require('../dbHelper').Prices;
...
+    }
+
+    try {
+        return await Products.create(productData, { include: [Products.Price], as: 'personId'});
+    } catch (error) {
+        throw new Error(error);
+    }
 };
 
+//TODO
```
Comment (bionanek) [inline api/db/repositories/productRepository.js]: is this comment needed here?

Comment (bionanek) [reply api/db/repositories/productRepository.js]: if it is then add a description of what has to be done `//TODO: change Promise to async below` for example

File: api/db/repositories/productRepository.js [hunk 2]
```diff
@@ -31,6 +32,7 @@ exports.updateProduct = (productId, updatedProduct) => {
     });
 };
 
+//TODO
```
Comment (bionanek) [inline api/db/repositories/productRepository.js]: .

File: api/db/repositories/productRepository.js [hunk 3]
```diff
@@ -59,31 +61,34 @@ exports.deleteProduct = (productId) => {
...
-            })
-            .catch((error) => {
-                reject(error)
-            });
-    });
+exports.getAll = async () => {
+    let allProducts;
+
+    try {
+        allProducts = await Products.findAll({ include: [Products.Price] }).map(el => el.get({ plain: true }));
```
Comment (bionanek) [inline api/db/repositories/productRepository.js]: can you break this line so it's not that long and easier to read?

File: api/db/repositories/productRepository.js [hunk 4]
```diff
@@ -59,31 +61,34 @@ exports.deleteProduct = (productId) => {
...
+    } catch (error) {
+        throw new Error(error);
+    }
+
+    if (allProducts === null || allProducts === undefined) {
+        throw new Error('Products table is empty. REPO');
+    } else {
```
Comment (bionanek) [inline api/db/repositories/productRepository.js]: .

File: api/db/repositories/productRepository.js [hunk 5]
```diff
@@ -59,31 +61,34 @@ exports.deleteProduct = (productId) => {
...
-    });
+    try {
+        requestedProduct = await Products.findById(productId, { include: [Products.Price] });
+    } catch (error) {
+        throw new Error(error);
+    }
+
+    if (requestedProduct === null || requestedProduct === undefined) {
+        throw new Error('Product with given ID doesn\'t exist')
+    } else {
```
Comment (bionanek) [inline api/db/repositories/productRepository.js]: .

File: api/domain/productsDomain.js [hunk 6]
```diff
@@ -21,66 +22,55 @@ exports.create = (productData) => {
...
-    return new Promise((resolve, reject) => {
-        productRepo.updateProduct(productId, updatedProduct)
-            .then((affectedRows) => {
-                resolve(affectedRows);
-            })
-            .catch((error) => {
-                reject(error);
-            });
-    });
+//MUST CHECK
```
Comment (bionanek) [inline api/domain/productsDomain.js]: what's that about?

File: api/domain/productsDomain.js [hunk 7]
```diff
@@ -21,66 +22,55 @@ exports.create = (productData) => {
...
+exports.update = async (productId, updatedProductData) => {
+    if (updatedProductData === null) {
+        throw new Error('Provide values to update.');
+    }
+
+    let affectedRows = await productRepo.updateProduct(productId, updatedProductData);
+
+    if (affectedRows == 0) {
+        throw new Error('No product updated.');
+    } else {
```
Comment (bionanek) [inline api/domain/productsDomain.js]: .

File: api/domain/productsDomain.js [hunk 8]
```diff
@@ -21,66 +22,55 @@ exports.create = (productData) => {
...
-            })
-            .catch((error) => {
-                reject(error);
-            });
-    });
+    let affectedRows = await productRepo.deleteProduct(productId);
+
+    if (affectedRows === 0) {
+        throw new Error('No product deleted.');
+    } else {
```
Comment (bionanek) [inline api/domain/productsDomain.js]: .

File: api/domain/productsDomain.js [hunk 9]
```diff
@@ -21,66 +22,55 @@ exports.create = (productData) => {
...
 
-                resolve(products);
-            })
-            .catch((error) => {
-                reject(error);
-            });
-    });
+    if (fetchedRows === 0) {
+        throw new Error('No products found.');
+    } else {
```
Comment (bionanek) [inline api/domain/productsDomain.js]: .

File: api/domain/productsDomain.js [hunk 10]
```diff
@@ -21,66 +22,55 @@ exports.create = (productData) => {
...
-            })
-            .catch((error) => {
-                reject(error);
-            });
-    });
+    let fetchedRow = await productRepo.getById(productId);
+
+    if (fetchedRow === 0) {
+        throw new Error('No product found.');
+    } else {
```
Comment (bionanek) [inline api/domain/productsDomain.js]: .

File: api/db/repositories/productRepository.js (hunk 1 above)
Comment (elvixpix) [reply api/db/repositories/productRepository.js]: Roger

File: api/db/repositories/productRepository.js (hunk 2 above)
Comment (elvixpix) [reply api/db/repositories/productRepository.js]: that

File: api/db/repositories/productRepository.js (hunk 3 above)
Comment (elvixpix) [reply api/db/repositories/productRepository.js]: Can try

File: api/db/repositories/productRepository.js (hunk 4 above)
Comment (elvixpix) [reply api/db/repositories/productRepository.js]: k

File: api/db/repositories/productRepository.js (hunk 5 above)
Comment (elvixpix) [reply api/db/repositories/productRepository.js]: k

File: api/domain/productsDomain.js (hunk 6 above)
Comment (elvixpix) [reply api/domain/productsDomain.js]: need help

File: api/domain/productsDomain.js (hunk 7 above)
Comment (elvixpix) [reply api/domain/productsDomain.js]: k

File: api/domain/productsDomain.js (hunk 8 above)
Comment (elvixpix) [reply api/domain/productsDomain.js]: k

File: api/domain/productsDomain.js (hunk 9 above)
Comment (elvixpix) [reply api/domain/productsDomain.js]: k

File: api/domain/productsDomain.js (hunk 10 above)
Comment (elvixpix) [reply api/domain/productsDomain.js]: k

**Comment by elvixpix** [comment]
Still working on updateProduct function in Repo.

File: api/db/repositories/productRepository.js [hunk 11]
```diff
@@ -3,22 +3,18 @@ const Prices = require('../dbHelper').Prices;
...
+exports.createProduct = async (productData) => {
+    try {
+        return await Products.create(productData, { include: [Products.Price], as: 'productId'});
+    } catch (error) {
+        throw new Error(error);
+    }
 };
 
-exports.updateProduct = (productId, updatedProduct) => {
+exports.updateProduct = (productId, updatedProductData) => {
```
Comment (bionanek) [inline api/db/repositories/productRepository.js]: this one is not converted yet

File: api/db/repositories/productRepository.js [hunk 12]
```diff
@@ -31,17 +27,34 @@ exports.updateProduct = (productId, updatedProduct) => {
...
-    return new Promise((resolve, reject) => {
-        let affectedRows = 0;
-        let priceId;
-        
-        Products.findById(productId)
-            .then((product) => {
-                if (product === null || product === undefined) {
-                    return reject({ message: 'Product with given ID doesn\'t exist' });
-                }
+// exports.updateProduct = async (productId, updatedProductData) => {
```
Comment (bionanek) [inline api/db/repositories/productRepository.js]: remove commented code

File: api/db/repositories/productRepository.js [hunk 13]
```diff
@@ -51,39 +64,42 @@ exports.deleteProduct = (productId) => {
             })
             .then((removedRows) => {
                 affectedRows += removedRows;
-                resolve(affectedRows);
+                return affectedRows;
```
Comment (bionanek) [inline api/db/repositories/productRepository.js]: if I think about it I suppose you can remove `affectedRows` variable and just return `removedRows` ?

**Pull request merged**